cql3/selection/selection.cc

```diff
@@ -659,6 +659,20 @@ bool result_set_builder::restrictions_filter::do_filter(const selection& selecti
         }
     }
 
+    if (_clustering_key_filter) {
+        if (!expr::is_satisfied_by(
+                        *_clustering_key_filter,
+                        expr::evaluation_inputs{
+                            .partition_key = partition_key,
```

Comment:
Again I think the real partition_key doesn't matter. But I'm starting to think maybe you want to combine the two expressions later, so you'll want to pass both partition and clustering key correctly. I'll wait and see...

Reply:
I see at the end you didn't combine the partition-key expression and the clustering-key expression (you created a "partition level filter" vs a "row level filter") but I agree that eventually we may want to have just one filter (e.g., imagine we allow the syntax "where p=c", this is a row-level filter but needs to read the partition key.
So I agree, your code is fine.

Reply:
We'll still want to have two filters (as an optimization). p=c will be moved to the per-clustering-row filter (even though it references a partition key), because its value is not constant across a partition.

This is standard loop-invariant code motion optimization.